docs/writing-docs/autodocs.md

```diff
@@ -55,16 +55,41 @@ By default, Storybook offers zero-config support for documentation and automatic
 
 <!-- prettier-ignore-end -->
 
-| Option        | Description                                                                                      |
-| ------------- | ------------------------------------------------------------------------------------------------ |
-| `disable`     | Toggles support for all documentation pages <br/> `docs: { disable:true }`                       |
-| `autodocs`    | Disables auto-generated documentation pages created via `tags` <br/> `docs: { autodocs: false }` |
-| `true`        | Enables auto-generated documentation pages for every component <br/> `docs: { autodocs: true }`  |
-| `defaultName` | Renames the auto-generated documentation page<br/> `docs: { defaultName: 'Documentation' }`      |
+| Option        | Description                                                                                                                          |
+| ------------- | ------------------------------------------------------------------------------------------------------------------------------------ |
+| `disable`     | Toggles support for all documentation pages <br/> `docs: { disable:true }`                                                           |
+| `autodocs`    | Configures auto-generated documentation pages. Available options: `true`, `false`,`tags` (default) <br/> `docs: { autodocs: false }` |
+| `defaultName` | Renames the auto-generated documentation page<br/> `docs: { defaultName: 'Documentation' }`                                          |
 
 ### Write a custom template
 
-To replace the default documentation template used by Storybook, you can create your own template written in MDX and update your UI configuration file (i.e., `.storybook/preview.js`) to point to it.
+To replace the default documentation template used by Storybook, you can extend your UI configuration file (i.e., `.storybook/preview.js`) and introduce a `docs` [parameter](../writing-stories/parameters.md#global-parameters). This parameter accepts a `page` function that returns a React component, which you can use to generate the required template. For example:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-preview-auto-docs-custom-template-function.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<div class="aside">
+💡 Internally, Storybook uses a similar implementation to generate the default template. See the Doc Blocks API reference if you want to learn more about how Doc Blocks work.
```

Comment:
Let's add a link for the API reference and coordinate this PR merging with https://github.com/storybookjs/storybook/pull/21113